posting/index.go

```diff
@@ -595,33 +650,74 @@ func needsIndexRebuild(old *pb.SchemaUpdate, current *pb.SchemaUpdate) indexOp {
 		currTokens[t] = true
 	}
 
+	// If the tokenizers are the same, nothing needs to be done.
 	if equal := reflect.DeepEqual(prevTokens, currTokens); equal {
-		return indexNoop
+		return indexRebuildInfo{
+			op: indexNoop,
+		}
+	}
+
+	// Else, the tokenizers from the old schema that do not exist in the schema
+	// need to be deleted and the tokenizers from the current schema that did
+	// not exist in the old one need to be built.
+	tokenizersToDelete := make([]string, 0)
+	for token := range prevTokens {
```

Comment:
If we change the prevTokens and currTokens from map[string]bool to map[string]struct{} (since they are essentially sets), then we can reuse the x.Diff method to get the differences.